### ISSUE TITLE
Basic Authentication is turned on even if we don't define the options.auth property

### ISSUE THREAD

**Issue by erichans** [victorherraiz/cloud-config-client#33]
Hi guys,

I'm trying to use this cloud-config-client and my server endpoints are open with no authentication.
When I try to configure using version 1.5.0+ I get HTTP 401 because of the code below:

```
function getAuth (auth, url) {
  if (auth && auth.user && auth.pass) {
    return auth.user + ':' + auth.pass
  }
  return url.username + ':' + url.password
}

```

It returns the string ":" even when there is no options.auth property defined anywhere because of the last return and it triggers the http Basic Authentication with empty username and password.

I'd suggest to change the coode above to something like the one below to avoid the problem:

```
function getAuth (auth, url) {
  if (auth && auth.user && auth.pass) {
    return auth.user + ':' + auth.pass
  }
  if(url.username || url.password) {
    return url.username + ':' + url.password
  }
  return auth
}
```

Thanks in advance.

**Comment by erichans** [comment]
@victorherraiz, is this possible to be implemented in the next version?

**Comment by victorherraiz-santander** [comment]
This is a bug, I will release a new version in a couple of hours.

**Comment by victorherraiz-santander** [comment]
fix publish as 1.6.2
